src/odemis/gui/cont/tabs.py

```diff
@@ -7202,6 +7279,139 @@ def _on_autofocus_done(self, future):
         # Enable manual focus when done running autofocus
         self.panel.btn_manual_focus.Enable(True)
 
+    @call_in_wx_main
+    def _on_specswitch_align(self, event):
```

Comment:
I did the change (removing the decorator). See my explanation above. The decorator is not "for every function which adjust the GUI components". It's for "for every function which adjust the GUI components **and could be run from a separate thread**". In particular, functions to handle GUI events are **always** run in the main GUI thread. Secondly, this decorator causes the function to always run asynchronously. So the caller thinks the function is "finished" (and it will receive `None` as return value), but actually the function is only put in a queue to be called later. This causes issues with the WX event handler function. Therefore, we should never decorate such function.